cmd/main.go

```diff
@@ -270,11 +271,15 @@ func getOutputFn(outputFlag *string, currentFn selector.InstanceTypesOutputFn) s
 }
 
 func getRegionAndProfileAWSSession(regionName *string, profileName *string) (*session.Session, error) {
-	sessOpts := session.Options{}
+	sessOpts := session.Options{SharedConfigState: session.SharedConfigEnable}
 	if regionName != nil {
 		sessOpts.Config.Region = regionName
 	}
 
+	if awsProfileName, ok := os.LookupEnv(awsProfileEnvVar); profileName == nil && ok && awsProfileName != "" {
+		profileName = &awsProfileName
+	}
+
 	if profileName != nil {
 		sessOpts.Profile = *profileName
 		if sessOpts.Config.Region == nil {
```

Comment:
can you delete this block as well? I'm fairly certain this block is no longer needed since the SDK is now doing the work of populating a region in the right precedence.

Reply:
you are right; Just tested without the block, and read the https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#SharedConfigState carefully; Indeed not needed.